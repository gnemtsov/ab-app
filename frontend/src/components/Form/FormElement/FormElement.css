/**********Label**********/

.Label {
    font-weight: bold;
    text-align: left;
}

.Label.Horizontal {
    grid-column: 1 / 1;
}

@media (min-width: 40rem) {
    .Label.Horizontal {
        padding: 0.4rem;
        text-align: right;
        max-width: 16rem;
    }
}

.Label.Inline {
    padding: 0.4rem;
}

/**********Input**********/

.InputContainer.Horizontal {
    grid-column: 1 / 1;
}

.InputContainer.MouseTriggered.Inline {
    padding-top: 0.4rem;
}

@media (min-width: 40rem) {
    .InputContainer.Horizontal {
        grid-column: 2 / 2;
    }
    .InputContainer.MouseTriggered.Horizontal {
        padding-top: 0.4rem;
    }
}

.InputText,
.Select,
.Textarea {
    font-family: 'Arial', sans-serif;
    font-size: 1rem;
    padding-left: 0.4rem;
    outline: none;
    width: 100%;
    border: 1px solid #666;
    background-color: transparent;
    box-shadow: none;
}

.Inline .InputText,
.Inline .Select,
.Inline .Textarea,
.Inline .Unrecognized {
    width: 12rem;
}

.InputText {
    height: 2rem;
}

.InputText.Invalid,
.InputSelect.Invalid,
.Textarea.Invalid {
    border: 1px solid red;
    background-color: #FDA49A;
}

.Select {
    height: 2rem;
    padding-right: 1.2rem;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABoAAAAaCAMAAACelLz8AAAAJ1BMVEVmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmaP/QSjAAAADHRSTlMAAgMJC0uWpKa6wMxMdjkoAAAANUlEQVR4AeXJyQEAERAAsNl7Hf3X6xt0QL6JpZWq30pdvdadme+0PMdzvHm8YThHcT1H7K0BtOMDniZhWOgAAAAASUVORK5CYII=);
    -webkit-background-size: 1rem 1rem;
    background-size: 1rem;
    background-repeat: no-repeat;
    background-position: calc(100% - 0.1rem) 50%;
}

.Select option {
    font-family: inherit;
}

.Textarea {
    height: 6rem;
}

.InputCheck,
.InputRadio {
    outline: none;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    height: 1.2rem;
    width: 1.2rem;
    margin: 0;
}

.InputCheck {
    border: 1px solid #666;
}

.InputCheck:checked {
    background-position: center;
    background-size: 90%;
    background-image: url(data:image/svg+xml;base64,PHN2ZyAgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiAgd2lkdGg9IjI0IiAgaGVpZ2h0PSIyNCIgIHZpZXdCb3g9IjAgMCAyNCAyNCIgIGZpbGw9Im5vbmUiICBzdHJva2U9IiM2NjY2NjYiICBzdHJva2Utd2lkdGg9IjQiICBzdHJva2UtbGluZWNhcD0icm91bmQiICBzdHJva2UtbGluZWpvaW49InJvdW5kIj4gIDxwb2x5bGluZSBwb2ludHM9IjIwIDYgOSAxNyA0IDEyIiAvPjwvc3ZnPg==);
}

.InputRadio {
    margin-right: 0.4rem;
    background-position: center;
    background-size: contain;
    background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0IiBmaWxsPSJub25lIiBzdHJva2U9IiM2NjY2NjYiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIj48Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSIxMCI+PC9jaXJjbGU+PC9zdmc+);
}

.InputRadio:disabled {
	background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0IiBmaWxsPSIjZWVlZWVlIiBzdHJva2U9IiM2NjY2NjYiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIj48Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSIxMCI+PC9jaXJjbGU+PC9zdmc+);
}


.InputRadio:checked {
    background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0IiBmaWxsPSJub25lIiBzdHJva2U9IiM2NjY2NjYiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIj48Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSIxMCI+PC9jaXJjbGU+PGNpcmNsZSBmaWxsPSIjNjY2NjY2IiBjeD0iMTIiIGN5PSIxMiIgcj0iNSI+PC9jaXJjbGU+PC9zdmc+);
}

.InputRadio:checked:disabled {
	background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0IiBmaWxsPSIjZWVlZWVlIiBzdHJva2U9IiM2NjY2NjYiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIj48Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSIxMCI+PC9jaXJjbGU+PGNpcmNsZSBjeD0iMTIiIGN5PSIxMiIgcj0iNiIgZmlsbD0iIzY2NjY2NiI+PC9jaXJjbGU+PC9zdmc+);
}

.InputRadioContainer {
    Horizontal-align: top;
    white-space: nowrap;
}

.InputRadioContainer:first-child {
    margin-right: 1.5rem;
}

.InputText:disabled,
.Select:disabled,
.Textarea:disabled,
.InputCheck:disabled {
    background-color: #eee;
}

.Unrecognized {
    font-size: 1rem;
    display: inline-block;
    height: 2rem;
    width: 100%;
    padding: 0.4rem 0 0 0.4rem;
    color: red;
    border: 1px dashed red;
}

.Message {
    color: red;
    line-height: 0.75rem;
    font-size: 0.75rem;
    min-height: 0.75rem;
    margin: 0.1rem 0 0.75rem 0;
    width: 4rem;
    min-width: 100%;
}

/**********Icon**********/

.InfoIcon {
    padding-top: 0.1rem;
    width: 26px;
    height: 26px;
}

.InfoIcon.Horizontal {
    grid-column: 2 / 2;
}

.InfoIcon.Inline {
    margin-left: 0.5rem;
}

@media (min-width: 40rem) {
    .InfoIcon.Horizontal {
        grid-column: 3 / 3;
    }
}