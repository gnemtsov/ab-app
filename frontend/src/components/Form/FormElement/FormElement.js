import React from 'react';
import moment from 'moment';

import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker-cssmodules.css';

import classes from './FormElement.css';

const FormElement = (props) => {
    //layout type
    let layoutClass = props.layout === 'inline' ? classes.Inline : classes.Horizontal;

    //------->label
    const label =
        <label
            key={props.id + '_label'}
            className={classes.Label + ' ' + layoutClass + ' ' + ((props.labelStyle && props.labelStyle.leftText) ? classes.LeftText : '')}
            htmlFor={props.id}
            style={props.labelStyle ? {gridArea: props.labelStyle.gridArea} : undefined} >
            {props.label + (props.required ? '*' : '')}
        </label>;

    //------->input
    let inputContainerClasses = [classes.InputContainer, layoutClass];
    let inputClasses = [];
    if (props.message.length) {
        inputClasses.push(classes.Invalid);
    }
    if (props.bodyStyle && props.bodyStyle.leftText) {
		inputClasses.push(classes.LeftText);
	}
    let input = [];
    switch (props.type) {

        case 'String':
            if (props.allowedValues !== undefined && props.allowedValues !== null && props.allowedValues.length) {

                if (props.allowedValues.length === 2) { //radio buttons
                    inputContainerClasses.push(classes.MouseTriggered);
                    let radios = props.allowedValues.map(
                        (value, index) =>
                            <label
                                key={props.id + '_radio' + index}
                                className={classes.InputRadioContainer}>
                                <input
                                    type="radio"
                                    className={classes.InputRadio}
                                    id={props.id + index}
                                    value={value}
                                    disabled={props.disabled ? true : false}
                                    checked={value === props.value}
                                    onChange={props.inputChanged} />
                                {value}
                            </label>
                    );
                    input = input.concat(radios);

                } else { //select
                    inputClasses.push(classes.Select);
                    input.push(
                        <select
                            key={props.id + '_select'}
                            className={inputClasses.join(' ')}
                            id={props.id}
                            disabled={props.disabled ? true : false}
                            value={props.value}
                            onChange={props.inputChanged}>
                            {props.allowedValues.map((value, i) => <option key={i}>{value}</option>)}
                        </select>
                    );
                }

            } else { //input type="text"
                inputClasses.push(classes.InputText);
                input.push(
                    <input
                        key={props.id + '_input'}
                        type={props.noEcho ? "password" : "text"}
                        className={inputClasses.join(' ')}
                        id={props.id}
                        placeholder={props.placeholder}
                        value={props.value}
                        disabled={props.disabled ? true : false}
                        onChange={props.inputChanged} />
                );
            }
            break;

        case 'Boolean': //checkbox
            inputContainerClasses.push(classes.MouseTriggered);
            input.push(
                <input
                    key={props.id + '_check'}
                    type="checkbox"
                    className={classes.InputCheck}
                    id={props.id}
                    disabled={props.disabled ? true : false}
                    checked={Boolean(props.value)}
                    onChange={props.inputChanged} />
            );
            break;

        case 'Number': //input type="text"
            inputClasses.push(classes.InputText);
            input.push(
                <input
                    key={props.id + '_input'}
                    type={"text"}
                    className={inputClasses.join(' ')}
                    id={props.id}
                    placeholder={props.placeholder}
                    value={props.value}
                    disabled={props.disabled ? true : false}
                    onChange={props.inputChanged} />
            );
            break;

        case 'Text': //textarea
            inputClasses.push(classes.Textarea);
            input.push(
                <textarea
                    key={props.id + '_textarea'}
                    className={inputClasses.join(' ')}
                    id={props.id}
                    placeholder={props.placeholder}
                    value={props.value}
                    disabled={props.disabled ? true : false}
                    onChange={props.inputChanged} />
            );
            break;

        case 'Date': //input type="text"
            inputClasses.push(classes.InputText);
            input.push(
                <DatePicker
                    key={props.id + '_input'}
                    className={inputClasses.join(' ')}
                    selected={moment(props.value)}
                    onChange={props.inputChanged} />
            );
            break;

        case 'Hidden': //input type="hidden"
            input.push(
                <input
                    key={props.id + '_input'}
                    type={"hidden"}
                    id={props.id}
                    value={props.value} />
            );
            break;

        default:
            input.push(
                <span
                    key={props.id + '_unrecognized'}
                    className={classes.Unrecognized}>
                    Unrecognized type
                </span>
            );
    }

    const message =
        <div
            key={props.id + '_validatorMessage'}
            className={classes.Message}>
            {props.message}
        </div>;

    //------->info icon with description
    const icon =
        <div
            key={props.id + '_infoIcon'}
            className={classes.InfoIcon + ' ' + layoutClass}
            data-tip={props.description}>
            {props.description !== undefined && props.description.length ? props.infoIcon : null}
        </div>;

    //------->output
    if (props.type !== 'Hidden') {
        return ([
            label,
            <div
				style={props.bodyStyle ? {gridArea: props.bodyStyle.gridArea} : undefined}
                key={props.id + '_inputContainer'}
                className={inputContainerClasses.join(' ')}>
                {[input, message]}
            </div>,
            icon
        ]);
    } else {
        return ([
            <div key="dummyDiv1"></div>,
            input,
            <div key="dummyDiv2"></div>
        ]);
    }
};

export default FormElement;
