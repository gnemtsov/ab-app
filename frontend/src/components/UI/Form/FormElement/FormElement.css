.Label {
    font-weight: bold;
    line-height: 1.2rem;
    text-align: left;
    padding-bottom: 0.5rem;
}

@media (min-width: 40rem) {
    .Label {
        text-align: right;
        padding-top: 0.42rem;
        padding-bottom: 1rem;
        padding-right: 0.4rem;
    }
}

.InputContainer input[type="text"],
.InputContainer input[type="password"],
.InputContainer select,
.InputContainer textarea {
    font-family: 'Arimo', sans-serif;
    font-size: 1rem;
    outline: none;
    width: 100%;
    vertical-align: top;
    background-color: transparent;
    border: 1px solid #666;
}

.InputText {
    line-height: 2rem;
    padding: 0 0.3rem;
}

.Select {
    line-height: 2rem;
    padding-left: 0.3rem;
    padding-right: 1.2rem;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABoAAAAaCAMAAACelLz8AAAAJ1BMVEVmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmZmaP/QSjAAAADHRSTlMAAgMJC0uWpKa6wMxMdjkoAAAANUlEQVR4AeXJyQEAERAAsNl7Hf3X6xt0QL6JpZWq30pdvdadme+0PMdzvHm8YThHcT1H7K0BtOMDniZhWOgAAAAASUVORK5CYII=);
    -webkit-background-size: 1rem 1rem;
    background-size: 1rem;
    background-repeat: no-repeat;
    background-position: calc(100% - 0.1rem) 50%;
}

.Textarea {
    padding: 0.4rem 0.3rem;
    height: 8rem;
}

.InputCheck {
    height: 1.5rem;
    width: 1.3rem;
    margin: 0.25rem 0 0 0;
    vertical-align: top;
}

.InputRadioContainer {
    line-height: 1.5rem;
    margin-right: 1.5rem;
    white-space: nowrap;
    vertical-align: bottom;
}

.InputRadio {
    height: 1.5rem;
    width: 1.3rem;
    margin: 0.25rem 0.25rem 0 0;
    vertical-align: top;
}

.Unrecognized {
    display: inline-block;
    line-height: 2rem;
    width: 100%;
    padding: 0 0.3rem;
    vertical-align: top;
    color: red;
    font-size: 1rem;
    border: 1px dashed red;
}

.IconContainer svg {
    margin-top: 0.1rem;
}

.Message {
    color: red;
    line-height: 0.75rem;
    font-size: 0.75rem;
    min-height: 0.75rem;
    margin: 0.1rem 0 0.75rem 0;
}

.InputText.Invalid,
.InputSelect.Invalid,
.Textarea.Invalid {
    border: 1px solid red;
    background-color: #FDA49A;
}

.Tooltip {
    max-width: 24rem;
    background-color: #ccc !important;
    color: #222 !important;
    padding: 0.5rem 0.8rem !important;
    font-size: 0.8rem !important;
    line-height: 1rem !important;
}

.Tooltip::after {
    border-top-color: transparent !important;
    border-right-color: transparent !important;
    border-bottom-color: transparent !important;
    border-left-color: transparent !important;
}

/*Positioning elements in the form grid*/

.Label,
.InputContainer {
    grid-column: 1 / 1;
}

.IconContainer {
    grid-column: 2 / 2;
}

@media (min-width: 40rem) {
    .Label {
        grid-column: 1 / 1;
    }
    .InputContainer {
        grid-column: 2 / 2;
    }
    .IconContainer {
        grid-column: 3 / 3;
    }
}