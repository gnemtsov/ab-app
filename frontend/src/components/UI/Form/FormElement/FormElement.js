import React from 'react';

import classes from './FormElement.css';

const FormElement = (props) => {
    let element = [];
    let inputClasses = [classes.Input];

    if (props.message.length) {
        inputClasses.push(classes.Invalid);
    }

    switch (props.type) {

        case 'String':
            element.push(
                <label
                    key={props.id + '_label'}
                    className={classes.Label}
                    htmlFor={props.id}>
                    {props.label}
                </label>
            );
            element.push(
                <div key={props.id + '_input'}>
                    <input
                        type="text"
                        className={inputClasses.join(' ')}
                        id={props.id}
                        placeholder={props.placeholder}
                        value={props.value}
                        required={props.required ? true : false}
                        onChange={props.changed} />

                    <div
                        className={classes.Message}>
                        {props.message}
                    </div>
                </div>
            );
            break;

        case 'Boolean':
            element.push(
                <label
                    key={props.id + '_label'}
                    className={classes.Label}
                    htmlFor={props.id}>
                    {props.label}
                </label>
            );
            element.push(
                <div key={props.id + '_input'}>
                    <input
                        type="checkbox"
                        className={inputClasses.join(' ')}
                        id={props.id}
                        value={props.value ? 'on' : 'off'}
                        onChange={props.changed} />
                    <div
                        className={classes.Message}>
                        {props.message}
                    </div>
                </div>
            );
            break;

        /* 
                case ('textarea'):
                    inputElement = <textarea
                        className={inputClasses.join(' ')}
                        {...props.elementConfig}
                        value={props.value}
                        onChange={props.changed} />;
                    break;
                case ('select'):
                    inputElement = (
                        <select
                            className={inputClasses.join(' ')}
                            value={props.value}
                            onChange={props.changed}>
                            {props.elementConfig.options.map(option => (
                                <option key={option.value} value={option.value}>
                                    {option.displayValue}
                                </option>
                            ))}
                        </select>
                    );
                    break; */
        default:
            element = 'Undefined';
    }

    return element;

};

export default FormElement;