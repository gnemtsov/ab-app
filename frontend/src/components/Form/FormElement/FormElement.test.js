import React from 'react';

import { configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FormElement from './FormElement';

configure({ adapter: new Adapter() });

describe('FormElement', () => {
	it('select', () => {
		const wrapper = render(
			<FormElement
				type='String'
				value='Test'
				id='test'
				value='1'
				allowedValues={['1', '2', '3']}
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('select change', () => {
		let flag = false;
		const wrapper = mount(
			<FormElement
				type='String'
				value='Test'
				id='test'
				value='1'
				allowedValues={['1', '2', '3']}
				message=''
				inputChanged={() => {flag = true;}}
			/>
		);
		wrapper.find('select').simulate('change');
		expect(flag).toEqual(true);
	});
	
	it('radio', () => {
		const wrapper = render(
			<FormElement
				type='String'
				value='Test'
				id='test'
				value='1'
				allowedValues={['1', '2']}
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('text', () => {
		const wrapper = render(
			<FormElement
				type='String'
				value='Test'
				id='test'
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});
	
	it('password', () => {
		const wrapper = render(
			<FormElement
				type='String'
				value='Test'
				noEcho
				id='test'
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});
	
	it('checkbox', () => {
		const wrapper = render(
			<FormElement
				type='Boolean'
				id='test'
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});
	
	it('number', () => {
		const wrapper = render(
			<FormElement
				type='Number'
				id='test'
				value={5}
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});
	
	it('date', () => {
		const wrapper = render(
			<FormElement
				type='Date'
				id='test'
				value='2005-05-05'
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});
	
	it('hidden', () => {
		const wrapper = render(
			<FormElement
				type='Hidden'
				id='test'
				value='hidden'
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});	
	
	it('unrecognized', () => {
		const wrapper = render(
			<FormElement
				type='ABCD'
				id='test'
				message=''
				inputChanged={() => {}}
			/>
		);
		expect(wrapper).toMatchSnapshot();		
	});
});
