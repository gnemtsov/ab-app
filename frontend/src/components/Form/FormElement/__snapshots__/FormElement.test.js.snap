// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormElement checkbox 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal MouseTriggered"
>
    <input
        class="InputCheck"
        id="test"
        type="checkbox"
    >
        
    </input>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement date 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal"
>
    <div>
        <div
            class="react-datepicker-wrapper"
        >
            <div
                class="react-datepicker__input-container"
            >
                <input
                    class="InputText"
                    type="text"
                    value="05/05/2005"
                >
                    
                </input>
            </div>
        </div>
    </div>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement hidden 1`] = `
Array [
  <div>
    
</div>,
  <input
    id="test"
    type="hidden"
    value="hidden"
>
    
</input>,
  <div>
    
</div>,
]
`;

exports[`FormElement number 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal"
>
    <input
        class="InputText"
        id="test"
        type="text"
        value="5"
    >
        
    </input>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement password 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal"
>
    <input
        class="InputText"
        id="test"
        type="password"
        value="Test"
    >
        
    </input>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement radio 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal MouseTriggered"
>
    <label
        class="InputRadioContainer"
    >
        <input
            checked=""
            class="InputRadio"
            id="test0"
            type="radio"
            value="1"
        >
            
        </input>
        1
    </label>
    <label
        class="InputRadioContainer"
    >
        <input
            class="InputRadio"
            id="test1"
            type="radio"
            value="2"
        >
            
        </input>
        2
    </label>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement select 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal"
>
    <select
        class="Select"
        id="test"
    >
        <option
            selected=""
        >
            1
        </option>
        <option>
            2
        </option>
        <option>
            3
        </option>
    </select>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement text 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal"
>
    <input
        class="InputText"
        id="test"
        type="text"
        value="Test"
    >
        
    </input>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;

exports[`FormElement unrecognized 1`] = `
Array [
  <label
    class="Label Horizontal "
    for="test"
>
    undefined
</label>,
  <div
    class="InputContainer Horizontal"
>
    <span
        class="Unrecognized"
    >
        Unrecognized type
    </span>
    <div
        class="Message"
    >
        
    </div>
</div>,
  <div
    class="InfoIcon Horizontal"
>
    
</div>,
]
`;
